client/src/components/pages/LandingPage.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Grid, Tabs, Tab, Box } from '@mui/material';
+
+import SignIn from '../auth/SignIn';
+import SignUp from '../auth/SignUp';
+
+function TabPanel(props) {
+  const { children, value, ...other } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      {...other}
+    >
+      <Box sx={{ p: 3 }}>{children[value]}</Box>
+    </div>
+  );
+}
+
+TabPanel.propTypes = {
+  children: PropTypes.node,
+  value: PropTypes.number.isRequired,
+};
+
+export const LandingPage = () => {
+  const [value, setValue] = React.useState(0);
+
+  const handleChange = (event, newValue) => {
+    setValue(newValue);
+  };
```

Comment:
Check out the [useCallback hook](https://reactjs.org/docs/hooks-reference.html#usecallback)

```suggestion
  const handleChange = useCallback((event, newValue) => {
    setValue(newValue);
  }, [setValue]);
```
Everytime LandingPage renders, the handle change is recreated and a new Tabs component will be returned because technically it has received a new prop

Reply:
Ok. I've looked into them. `useCallback` memoizes the callback, which caches the results from distinct inputs. Thanks!
I have a couple questions to dig a little deeper.

1. Should this always be used with callbacks?
2. In this particular example, does it matter to use the hook? Changing the state isn't that expensive right? Or is it safe to assume that the overhead for memoizing is always less than re-computing the callback?
3. Can this be used to memoize the API calls?

Reply:
I answered my questions by reading [this](https://dmitripavlutin.com/dont-overuse-react-usecallback/), but I'm still interested in your take if you want to share.

Reply:
I really like the summary of that article. I want to do that with this app later on.